public/main.js

```diff
@@ -61,12 +61,14 @@ const getPosts = () => {
           imageUrl: metadata.imageUrl ? metadata.imageUrl : "No image given",
           content: content ? content : "No content given",
         };
-
+        
         postlist.push(post);
+        console.log(`length of postlist: ${postlist.length}`)
         // only if you are at the LAST iteration, write the postlist array into a JSON object
-        if (i === files.length - 1) {
+        // compare length of postlist array to length of files array 
+        if (postlist.length === files.length) {
```

Comment:
Neat! I should have thought of this... what I don't understand is why the condition on line 67 doesn't work. It looks logical and clean and it clearly worked for you. Any hints as to why that doesn't work?

Reply:
If you log `i` to the console, you'll see that due to asynchronous behavior, the values are not logged in order. To be honest, I don't remember what I was thinking when I made the file. I was planning on doing something different, but it seemed beyond the scope of the tutorial, so I changed things up. I got lucky, I guess.